src/Illuminate/Mail/Mailable.php

```diff
@@ -877,18 +877,40 @@ public function attach($file, array $options = [])
         if ($file instanceof Attachable) {
             $file = $file->toMailAttachment();
         }
-
+    
         if ($file instanceof Attachment) {
             return $file->attachTo($this);
         }
-
+    
         $this->attachments = collect($this->attachments)
-                    ->push(compact('file', 'options'))
-                    ->unique('file')
-                    ->all();
+            ->push(compact('file', 'options'))
+            ->unique('file')
+            ->all();
 
         return $this;
     }
+    
+    /**
+     * Attach multiple files to the message.
+     *
+     * @param  array|object  $files
+     * @param  array  $options
+     * @return $this
+     */
+    public function attachMany($files, array $options = [])
+    {
+        if(is_array($files))
+        {
+            foreach ($files as $file => $options)
+            {
+                $this->attach((is_int($file) ? $options : $file), (is_int($file) ? [] : $options));
+            }
+        } else {
+            $this->attach($files, $options = []);
```

Comment:
You're forcing `$options` to always pass through an empty array here. But his method should probably force you to include an array or iterable object to not even have this "else" path, So if `$files` is null or empty an early return would probably be best.

Reply:
This should be addressed with the current changes.